compiler-core/src/language_server/server.rs

```diff
@@ -455,15 +456,47 @@ fn diagnostic_to_lsp(diagnostic: Diagnostic) -> Vec<lsp::Diagnostic> {
         .location
         .expect("Diagnostic given to LSP without location");
     let line_numbers = LineNumbers::new(&location.src);
+    let path = path_to_uri(location.path);
+    let range = src_span_to_lsp_range(location.label.span, &line_numbers);
+
+    let related_info = location
+        .extra_labels
+        .iter()
+        .map(|extra| {
+            let message = extra.label.text.clone().unwrap_or_default();
+            let location = if let Some(path) = &extra.path {
+                let line_numbers = LineNumbers::new(
+                    &extra
+                        .src
+                        .clone()
+                        .expect("different path must have source attached"),
```

Comment:
If these should be given together than they should be in the same `Option`

Reply:
Yeah makes sense I'll combine them into 1 tuple